crawl-ref/source/ouch.cc

```diff
@@ -653,33 +653,56 @@ static void _powered_by_pain(int dam)
     const int level = player_mutation_level(MUT_POWERED_BY_PAIN);
 
     if (you.mutation[MUT_POWERED_BY_PAIN]
-        && (random2(dam) > 4 + div_rand_round(you.experience_level, 4)
-            || dam >= you.hp_max / 2))
+        && (one_chance_in(5)
+            || dam >= you.hp_max / (2 + level)))
     {
-        switch (random2(4))
+        mpr("You focus on the pain.");
+        bool success = false;
+        int tries = 0;
+        do
         {
-        case 0:
-        case 1:
-        {
-            if (you.magic_points < you.max_magic_points)
+            tries++;
```

Comment:
rather than trying a random case each time, is there some way to randomly order the options and try them in turn? there's an annoying case with my code where 1/4 options is available but it doesn't get tried due to RNG